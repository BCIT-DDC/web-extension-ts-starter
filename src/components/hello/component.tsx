import React from "react";

// // // //

export function Hello() {
    return (
        <div className="row">
            <div className="col-lg-12 text-center">
                <p className="lead mb-0">Example Extension</p>
            </div>
        </div>
    );
}
