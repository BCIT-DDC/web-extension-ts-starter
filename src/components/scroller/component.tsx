import React from "react";
import css from "./styles.module.css";

// // // //

/**
 * Component that renders buttons to scroll to the top and bottom of the page
 */
export function Scroller(props: {
    onClickScrollTop: () => void;
    onClickScrollBottom: () => void;
}) {
    return (
        <div className="grid gap-3 grid-cols-2 mt-3 w-full">
            <button
                className={css.btn}
                data-testid="scroll-to-top"
                onClick={() => props.onClickScrollTop()}
            >
                Scroll To Top
            </button>
            <button
                className={css.btn}
                data-testid="scroll-to-bottom"
                onClick={() => props.onClickScrollBottom()}
            >
                Scroll To Bottom
            </button>
        </div>
    );
}
