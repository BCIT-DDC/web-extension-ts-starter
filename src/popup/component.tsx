import React from "react";
import { Hello } from "@src/components/hello";
import { browser, Tabs } from "webextension-polyfill-ts";
import { Scroller } from "@src/components/scroller";
import css from "./styles.module.css";

// // // //

// Scripts to execute in current tab
const scrollToTopScript = `window.scroll(0,0)`;
const scrollToBottomScript = `window.scroll(0,9999999)`;

/**
 * Executes a string of Javascript on the current tab
 * @param code The string of code to execute on the current tab
 */
function executeScript(code: string): void {
    // Query for the active tab in the current window
    browser.tabs
        .query({ active: true, currentWindow: true })
        .then((tabs: Tabs.Tab[]) => {
            // Pulls current tab from browser.tabs.query response
            const currentTab: Tabs.Tab | undefined = tabs[0];

            // Short circuits function execution is current tab isn't found
            if (!currentTab) {
                return;
            }

            // Executes the script in the current tab
            browser.tabs
                .executeScript(currentTab.id, {
                    code,
                })
                .then(() => {
                    console.log("Done Scrolling");
                });
        });
}

// // // //

export function Popup() {
    // Sends the `popupMounted` event
    React.useEffect(() => {
        browser.runtime.sendMessage({ popupMounted: true });
    }, []);

    // Renders the component tree
    return (
        <div className={css.popupContainer}>
            <div className="mx-4 my-4">
                <Hello />
                <hr />
                <Scroller
                    onClickScrollTop={() => {
                        executeScript(scrollToTopScript);
                    }}
                    onClickScrollBottom={() => {
                        executeScript(scrollToBottomScript);
                    }}
                />
            </div>
        </div>
    );
}
