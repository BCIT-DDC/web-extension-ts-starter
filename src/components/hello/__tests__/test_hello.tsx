import * as React from "react";
import { Hello } from "../component";
import renderer from "react-test-renderer";

it("component renders", () => {
    const tree = renderer.create(<Hello />).toJSON();
    expect(tree).toMatchSnapshot();
});
