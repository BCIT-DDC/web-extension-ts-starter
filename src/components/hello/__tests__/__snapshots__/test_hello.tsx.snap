// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-12 text-center"
  >
    <p
      className="lead mb-0"
    >
      Example Extension
    </p>
  </div>
</div>
`;
